test/e2e_federation/upgrade.go

```diff
@@ -25,9 +25,7 @@ import (
 	. "github.com/onsi/ginkgo"
 )
 
-var upgradeTests = []upgrades.Test{
-	upgrades.SecretUpgradeTest(),
-}
+var upgradeTests = upgrades.GenericUpgradeTests()
```

Comment:
I guess I see this as a case of YAGNI. If/when we get different kinds of upgrade tests, we can rename this function.  But for now it's just a way to separate creation of upgrade tests from their execution, and I'm not sure it's worth bikeshedding over the name of a function that is only ever intended to be called once.